L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc

```diff
@@ -0,0 +1,339 @@
+  // to do so I take a point, and look if is greater than the points around it (in the 9x9 neighborhood)
+  // to prevent mutual exclusion, I check greater or equal for points above and right to the one I am considering (including the top-left point)
+  // to prevent mutual exclusion, I check greater for points below and left to the one I am considering (including the bottom-right point)
+
+  for (int iPhi = 1; iPhi <= nBinsY; iPhi++) {
+    for (int iEta = 1; iEta <= nBinsX; iEta++) {
+      float centralPt = caloGrid.GetBinContent(iEta, iPhi);
+      if (centralPt < seedThreshold)
+        continue;
+      bool isLocalMaximum = true;
+
+      // Scanning through the grid centered on the seed
+      for (int etaIndex = -etaHalfSize; etaIndex <= etaHalfSize; etaIndex++) {
+        for (int phiIndex = -phiHalfSize; phiIndex <= phiHalfSize; phiIndex++) {
+          if ((etaIndex == 0) && (phiIndex == 0))
+            continue;
+          if (phiIndex > 0) {
+            if (phiIndex > -etaIndex) {
+              isLocalMaximum =
+                  ((isLocalMaximum) && (centralPt > getTowerEnergy(caloGrid, iEta + etaIndex, iPhi + phiIndex)));
+            } else {
+              isLocalMaximum =
+                  ((isLocalMaximum) && (centralPt >= getTowerEnergy(caloGrid, iEta + etaIndex, iPhi + phiIndex)));
+            }
+          } else {
+            if (phiIndex >= -etaIndex) {
+              isLocalMaximum =
+                  ((isLocalMaximum) && (centralPt > getTowerEnergy(caloGrid, iEta + etaIndex, iPhi + phiIndex)));
+            } else {
+              isLocalMaximum =
+                  ((isLocalMaximum) && (centralPt >= getTowerEnergy(caloGrid, iEta + etaIndex, iPhi + phiIndex)));
+            }
+          }
+        }
+      }
+      if (isLocalMaximum) {
+        seeds.emplace_back(std::make_tuple(iEta, iPhi));
+      }
+    }
+  }
+
+  return seeds;
+}
+
+reco::CaloJet Phase1L1TJetProducer::buildJetFromSeed(const TH2F& caloGrid, const std::tuple<int, int>& seed) {
+  int iEta = std::get<0>(seed);
+  int iPhi = std::get<1>(seed);
+
+  int etaHalfSize = (int)jetIEtaSize_ / 2;
+  int phiHalfSize = (int)jetIPhiSize_ / 2;
+
+  float ptSum = 0;
+  // Scanning through the grid centered on the seed
+  for (int etaIndex = -etaHalfSize; etaIndex <= etaHalfSize; etaIndex++) {
+    for (int phiIndex = -phiHalfSize; phiIndex <= phiHalfSize; phiIndex++) {
+      ptSum += getTowerEnergy(caloGrid, iEta + etaIndex, iPhi + phiIndex);
+    }
+  }
+
+  // Creating a jet with eta phi centered on the seed and momentum equal to the sum of the pt of the components
+  reco::Candidate::PolarLorentzVector ptVector;
+  ptVector.SetPt(ptSum);
+  ptVector.SetEta(caloGrid.GetXaxis()->GetBinCenter(iEta));
+  ptVector.SetPhi(caloGrid.GetYaxis()->GetBinCenter(iPhi));
+  reco::CaloJet jet;
+  jet.setP4(ptVector);
+  return jet;
+}
+
+std::vector<reco::CaloJet> Phase1L1TJetProducer::_buildJetsFromSeedsWithPUSubtraction(
```

Comment:
function can be `const` etc.

Reply:
done and removed `caloGrid` input: https://github.com/cms-l1t-offline/cmssw/blob/fc3a1a454f40efdf7c95dd97295a6a085a674092/L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc#L290-L291